sound/soc/sof/intel/hda-dai.c

```diff
@@ -151,77 +151,67 @@ static int hda_link_dma_params(struct hdac_ext_stream *stream,
 	return 0;
 }
 
-/* find the snd_sof_dai matching the BE CPU DAI name and update its config */
-static struct snd_sof_dai *hda_get_sof_dai(struct sof_intel_hda_stream *hda_stream,
-					   const char *dai_name, int channel, int dir)
+/* Update config for the DAI widget */
+static struct sof_ipc_dai_config *hda_dai_update_config(struct snd_soc_dapm_widget *w,
+							int channel)
 {
-	struct snd_sof_dev *sdev = hda_stream->sdev;
+	struct snd_sof_widget *swidget = w->dobj.private;
```

Comment:
w is converted from dai->playback_widget or dai->capture_widget and it is possible that dai->playback_widget or dai->capture_widget are not set, so w = NULL. please check w != NULL here to avoid kernel panic. Thanks

Reply:
@RanderWang this is not likely to happen with SOF DAIs because of the way we connect the CPU DAI with the DAIwidgets in sof_connect_dai_widget().

Reply:
not likely or not possible @ranj063 ? in the first case a bit of defensive programming wouldn't hurt. If we know however it cannot possible happen then we can move on.

Reply:
If there's no playback/capture widget associated with a BE CPU DAI, there wouldnt be a valid audio path. So I cant imagine the possibility of w being NULL ever.

Reply:
thanks, I checked the code.